src/Microsoft.AspNet.Routing/Infrastructure/PathSegment.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNet.Routing.Infrastructure
```

Comment:
This sounds like something that should really go in a `.Internal` namespace because we don't intend anyone ever to use this outside of Routing's own needs. Might need to do a Routing API review...